examples/controlnet/train_controlnet_flax.py

```diff
@@ -907,10 +907,15 @@ def compute_loss(params, minibatch, sample_rng):
 
             if args.snr_gamma is not None:
                 snr = jnp.array(compute_snr(timesteps))
-                if noise_scheduler.config.prediction_type == "v_prediction":
-                    # Velocity objective requires that we add one to SNR values before we divide by them.
-                    snr = snr + 1
-                snr_loss_weights = jnp.where(snr < args.snr_gamma, snr, jnp.ones_like(snr) * args.snr_gamma) / snr
+                if noise_scheduler.config.prediction_type == "epsilon":
+                    snr_loss_weights = jnp.where(snr < args.snr_gamma, snr, jnp.ones_like(snr) * args.snr_gamma) / snr
+                elif noise_scheduler.config.prediction_type == "v_prediction":
+                    snr_loss_weights = jnp.where(snr < args.snr_gamma, snr, jnp.ones_like(snr) * args.snr_gamma) / (
+                        snr + 1
+                    )
+                else:
+                    raise ValueError(f"Unknown prediction type {noise_scheduler.config.prediction_type}")
```

Comment:
technically it supported prediction_type = "sample" before, since the only special case was v-prediction.

should be able to put that back, as the prediction type check is earlier.

Reply:
I guess it is good to add another 'elif' branch here to take care of x0-prediction and leave the 'else' branch unchanged to allow for other potential prediction types in the future.